File: src/main/java/org/dedda/bratwurst/lang/Return.java
package org.dedda.bratwurst.lang;

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
public class Return extends BWExpression {

    private BWExpression expression;
    private BWObject value;

    public Return(int lineNumber, BWExpression expression) {
        super(lineNumber);
        this.expression = expression;
    }

    @Override
    public BWObject getValue() {
        return value;
    }

    @Override
    public int getIntValue() {
        return value.getIntValue();
    }

    @Override
    public String getValueType() {
        return value.getValueType();
    }

    @Override
    public void run(Scope scope) {
        expression.run(scope);
        value = expression.getValue();
    }
}

File: src/main/java/org/dedda/bratwurst/lang/Return.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
class Return(lineNumber: Int, private val expression: BWExpression) : BWExpression(lineNumber) {
    private var value: BWObject? = null
    override fun getValue(): BWObject {
        return value!!
    }

    override fun getIntValue(): Int {
        return value!!.intValue
    }

    override fun getValueType(): String {
        return value!!.valueType
    }

    override fun run(scope: Scope) {
        expression.run(scope)
        value = expression.value
    }

}